src/sentry/integrations/pipeline.py

```diff
@@ -34,6 +46,40 @@ def ensure_integration(key, data):
     return integration
 
 
+def is_violating_region_restriction(organization_id: int, integration_id: int):
+    """
+    Returns True if the organization_id provided does NOT reside within the same region as other
+    organizations which have installed the provided integration.
+    """
+    if SiloMode.get_current_mode() == SiloMode.MONOLITH:
+        return False
+
+    ois = OrganizationIntegration.objects.filter(integration_id=integration_id)
+    if len(ois) == 0:
+        return False
+
+    logger_extra = {
+        "integration_id": integration_id,
+        "organization_id": organization_id,
+    }
+
+    organization_ids = {oi.organization_id for oi in ois}
+    region_names = OrganizationMapping.objects.filter(
+        organization_id__in=organization_ids
+    ).values_list("region_name", flat=True)
+
+    if len(region_names > 1):
+        logger.error("region_violation", extra={"regions": region_names, **logger_extra})
```

Comment:
Couldn't you end up with multiple row here when the integration is installed in multiple organziations from the same region? I think you need a `set()` cast or a `.distinct()` on the query.